File: SurTerrain/src/Models/Reservation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Models;

import java.sql.Date;
import java.sql.Time;


/**
 *
 * @author OussKh
 */
public class Reservation {
    
    private int idRes;
    private Date date;
    private Time heureDebut;
    private Time heureFin;
    private int idTerrain;
    private int idClient;

    public Reservation() {
    }

    public Reservation(int idRes, Date date, Time heureDebut, Time heureFin, int idTerrain, int idClient) {
        this.idRes = idRes;
        this.date = date;
        this.heureDebut = heureDebut;
        this.heureFin = heureFin;
        this.idTerrain = idTerrain;
        this.idClient = idClient;
    }

    public Reservation(Date date, Time heureDebut, Time heureFin, int idTerrain, int idClient) {
        this.date = date;
        this.heureDebut = heureDebut;
        this.heureFin = heureFin;
        this.idTerrain = idTerrain;
        this.idClient = idClient;
    }

    public Reservation(Date date, Time heureDebut, Time heureFin, int idTerrain) {
        this.date = date;
        this.heureDebut = heureDebut;
        this.heureFin = heureFin;
        this.idTerrain = idTerrain;
    }
    

    public Reservation(Date date, Time heureDebut, Time heureFin) {
        this.date = date;
        this.heureDebut = heureDebut;
        this.heureFin = heureFin;
    }

    public int getIdRes() {
        return idRes;
    }

    public void setIdRes(int idRes) {
        this.idRes = idRes;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public Time getHeureDebut() {
        return heureDebut;
    }

    public void setHeureDebut(Time heureDebut) {
        this.heureDebut = heureDebut;
    }

    public Time getHeureFin() {
        return heureFin;
    }

    public void setHeureFin(Time heureFin) {
        this.heureFin = heureFin;
    }

    public int getIdTerrain() {
        return idTerrain;
    }

    public void setIdTerrain(int idTerrain) {
        this.idTerrain = idTerrain;
    }

    public int getIdClient() {
        return idClient;
    }

    public void setIdClient(int idClient) {
        this.idClient = idClient;
    }

  

    
    @Override
  public boolean equals(Object obj) {
         if (obj instanceof Reservation) {
            Reservation r = (Reservation) obj;
            if (r.idRes == this.idRes ) { 
                
                return true;
            }
        }
        return false;      
    }
   
    
    
    
}

File: src/Entity/Reservation.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Reservation
 *
 * @ORM\Table(name="reservation", indexes={@ORM\Index(name="fk_reservation", columns={"idTerrain"}), @ORM\Index(name="fkClient", columns={"idclient"})})
 * @ORM\Entity
 * @ORM\Entity(repositoryClass="App\Repository\ReservationRepository")
 */
class Reservation
{
    /**
     * @var int
     *
     * @ORM\Column(name="idRes", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idres;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date", type="date", nullable=false)
     */
    private $date;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="heureDebut", type="time", nullable=false)
     */
    private $heuredebut;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="heureFin", type="time", nullable=false)
     */
    private $heurefin;

    /**
     * @var \Client
     *
     * @ORM\ManyToOne(targetEntity="Client")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idclient", referencedColumnName="idC")
     * })
     */
    private $idclient;

    /**
     * @var \Terrain
     *
     * @ORM\ManyToOne(targetEntity="Terrain")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idTerrain", referencedColumnName="idTerrain")
     * })
     */
    private $idterrain;

    public function getIdres(): ?int
    {
        return $this->idres;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getHeuredebut(): ?\DateTimeInterface
    {
        return $this->heuredebut;
    }

    public function setHeuredebut(\DateTimeInterface $heuredebut): self
    {
        $this->heuredebut = $heuredebut;

        return $this;
    }

    public function getHeurefin(): ?\DateTimeInterface
    {
        return $this->heurefin;
    }

    public function setHeurefin(\DateTimeInterface $heurefin): self
    {
        $this->heurefin = $heurefin;

        return $this;
    }

    public function getIdclient(): ?Client
    {
        return $this->idclient;
    }

    public function setIdclient(?Client $idclient): self
    {
        $this->idclient = $idclient;

        return $this;
    }

    public function getIdterrain(): ?Terrain
    {
        return $this->idterrain;
    }

    public function setIdterrain(?Terrain $idterrain): self
    {
        $this->idterrain = $idterrain;

        return $this;
    }


}
